feat: new course kick off: Scraping with Apify and AI (#2275)
apify/apify-docs: sources/academy/platform/scraping_with_apify_and_ai/index.md

---
title: Scraping with Apify and AI
description: Learn how to use AI to extract information from websites in this practical course, starting from the absolute basics.
sidebar_position: 5
sidebar_label: Scraping with Apify and AI
category: apify platform
slug: /scraping-with-apify-and-ai
unlisted: true
---

import DocCardList from '@theme/DocCardList';

> **Learn how to use AI to extract information from websites in this practical course, starting from the absolute basics.**

Reviewer: ```suggestion
**Learn how to use AI to extract information from websites, starting from the absolute basics.**
```

---

> In this course we'll use AI assistants to create an application for watching prices. It'll be able to scrape product pages of an e-commerce website and record prices. Data from several runs of such program would be useful for seeing trends in price changes, detecting discounts, etc.

Reviewer: ```suggestion
In this practical course, we'll use AI assistants to create an application for watching prices. It'll be able to scrape product pages of an e-commerce website and record prices. Data gathered from several runs of such program would be useful for seeing trends in price changes or detecting discounts.
```

The end product will, unlike programs vibe-coded carelessly, reach the level of quality allowing for further extensibility and comfortable maintenance, so that it can be published to [Apify Store](https://apify.com/store).

## What we'll do

- Use ChatGPT (AI chat) to create a program which extracts data from a web page.

> - Save extracted data in various formats, e.g. CSV which MS Excel or Google Sheets can open.

Reviewer: ```suggestion
- Save extracted data in various formats, for example CSV which MS Excel or Google Sheets can open.
```

Author: Thanks for the fix! Is "for example" generally better than "e.g." in a text like this, or does it further depend on context? Is there a rule of thumb I can follow next time?

Reviewer: "For example" is just generally better, it improves readability and accessibility. Latin abbreviations might be confusing, especially for non-natives outside of "the West".

Author: Makes sense. Thanks!

- Use Cursor (AI agent) to improve the program so that it is robust and maintainable.

> - Save time and effort with Apify's scraping platform.

Reviewer: ```suggestion
- Save time and effort with the Apify platform.
```

> ## Who this course is for

Reviewer: ```suggestion
## Who is this course for
```

Author: Thanks, this mistake apparently survived copy-pasting for several times, so it's already in two other course intros in the Academy. I'll add a separate commit to address this and I'll change it everywhere.

Author: Done in f75b7649974f9118e7e5053d04ab5055fabec187

Anyone with basic knowledge of chatting with an AI assistant and affinity to building digital products who wants to start with web scraping can take this course. The course does not expect you to have any prior knowledge of web technologies or scraping.

## Requirements

> - Prior experience chatting with AI assistants, such as OpenAI ChatGPT, Google Gemini, or Anthropic Claude.

Reviewer: ```suggestion
- Prior experience chatting with AI assistants, such as ChatGPT from OpenAI, Google Gemini, or Claude by Anthropic.
```

- A macOS, Linux, or Windows machine. If it's your work computer, make sure you have permissions to install new software.

> - Familiarity with running commands in Terminal (macOS/Linux) or PowerShell (Windows). Just generally knowing what those are and how to use them is sufficient.

Reviewer: ```suggestion
- Familiarity with running commands in terminal (macOS/Linux) or PowerShell (Windows). Just knowing what those are and how to use them is sufficient.
```
